File: app/src/main/java/com/dominikgaller/alpinebooking/booking/core/domain/exception/BookingNotFoundException.java
package com.dominikgaller.alpinebooking.booking.core.domain.exception;

/**
 * Thrown when a booking with the given ID cannot be found in the repository.
 *
 * <p>Maps to HTTP 404 Not Found at the REST boundary.
 *
 * <p>SDD: See {@code documentation/use-cases/uc02-confirm-tour-booking.spec.md}, section 9.
 */
public class BookingNotFoundException extends RuntimeException {

    public BookingNotFoundException(final String bookingId) {
        super("Booking not found: " + bookingId);
    }
}

File: src/main/kotlin/com/example/service/booking/core/domain/tourbooking/exception/BookingNotFoundException.kt
package com.example.service.booking.core.domain.tourbooking.exception

/**
 * Thrown when no booking exists for a given identity.
 *
 * Maps to HTTP 404 (`coding-style.definition.md` § 6.2).
 */
class BookingNotFoundException(
    bookingId: String,
) : RuntimeException("Booking not found: $bookingId")
